kernel/init_static.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2012-2015 Wind River Systems, Inc.
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+void __do_global_ctors_aux(void);
+void __do_init_array_aux(void);
+
+void z_init_static(void)
+{
+#if defined(CONFIG_STATIC_INIT_GNU)
+	__do_global_ctors_aux();
+	__do_init_array_aux();
+#elif defined(__CCAC__) /* ARC MWDT */
+	__do_global_ctors_aux();
+#endif
+}
+
+/**
+ * @section - Constructor module
+ * @brief
+ * The ctors section contains a list of function pointers that execute both the
+ * C++ constructors of static global objects, as well as initializer functions
```

Comment:
Would be nice to clarify that this is not exclusive to C++ now.

Reply:
Is the description just following enough? or would you like something else?